scripts/prepare-crds

```diff
@@ -115,4 +115,165 @@ done
 # 	{{- required "Required CRDs are missing. Please install the ${name}-crd chart before installing this chart." "" -}}
 # {{- end -}}
 # {{- end -}}
-EOF
\ No newline at end of file
+EOF
+
+# Restructure the package to use Jobs to manage the CRD chart lifecyle instead of the templates/ directory if
+# the assumeOwnershipOfCRDs flag is enabled
+assume_ownership=$(cat ${f}/package.yaml | yq r - generateCRDChart.assumeOwnershipOfCRDs)
+if [[ "${assume_ownership}" != "true" ]]; then
+	exit 0
+fi
+
+mv ${f}/charts-crd/templates ${f}/charts-crd/crd-manifest
+mkdir -p ${f}/charts-crd/templates
+
+cat << EOF > ${f}/charts-crd/values.yaml
```

Comment:
Would it make sense to invest a little bit of time in creating something like a `chart-templates/` directory at the root of this repository with a `chart-templates/charts-crd/` and `chart-templates/chart-crd-assume-ownership` folder that has all these files within it? 

Using `cat << EOF > $PATH` to create all these files as part of a Bash script seems clunky, but I wasn't sure if it was worth investing the time to rewrite the `prepare-crd` script to do this (as opposed to maybe just doing it later when we switch to Go scripts)

Reply:
Sounds good, I can edit the PR to use that instead.

It shouldn't take too much longer, but it would be editing the logic of previously written code (as opposed to the current PR, which only add / modifies an additional set of files to the existing `prepare-crds` script). Should be relatively easy to do though.

Reply:
@cbron I added this change separately to the last commit on the PR. Let me know if the changes look good to you and I can accordingly squash them into my other commits.